integration-tests/tests/src/tests/results.ts

```diff
@@ -0,0 +1,1067 @@
+    openRealmBeforeEach({ schema: [TestObjectSchema] });
+
+    it("should have a valid constructor", function (this: RealmContext) {
+      const objects = this.realm.objects<TestObject>("TestObject");
+
+      expect(objects instanceof Realm.Results).equals(true);
+      expect(objects instanceof Realm.Collection).equals(true);
+
+      expect(() => {
+        new Realm.Results();
+      }).throws("Illegal constructor");
+
+      expect(typeof Realm.Results).equals("function");
+      expect(Realm.Results instanceof Function).equals(true);
+    });
+
+    it("should return the correct length", function (this: RealmContext) {
+      const objects = this.realm.objects("TestObject");
+      expect(objects.length).equals(0);
+
+      this.realm.write(() => {
+        this.realm.create("TestObject", { doubleCol: 1 });
+        expect(objects.length).equals(1);
+      });
+      expect(objects.length).equals(1);
+    });
+
+    it("implements isEmpty", function (this: RealmContext) {
+      const result = this.realm.objects("TestObject");
+      expect(result.isEmpty()).equals(true);
+      this.realm.write(() => {
+        this.realm.create("TestObject", { doubleCol: 42 });
+      });
+      expect(result.isEmpty()).equals(false);
+      this.realm.close();
+    });
+
+    it("implements indexOf", function (this: RealmContext) {
+      const [object1, object2, object3] = this.realm.write(() => {
+        return [
+          this.realm.create<TestObject>("TestObject", { doubleCol: 1 }),
+          this.realm.create<TestObject>("TestObject", { doubleCol: 2 }),
+          this.realm.create<TestObject>("TestObject", { doubleCol: 2 }),
+        ];
+      });
+
+      // Search in base table
+      const objects = this.realm.objects("TestObject");
+      expect(objects.indexOf(object1)).equals(0);
+      expect(objects.indexOf(object2)).equals(1);
+      expect(objects.indexOf(object3)).equals(2);
+
+      // Search in filtered query
+      const results = objects.filtered("doubleCol == 2");
+      expect(results.indexOf(object1)).equals(-1);
+      expect(results.indexOf(object2)).equals(0);
+      expect(results.indexOf(object3)).equals(1);
+
+      // Searching for object from the wrong realm
+      const realm2 = new Realm({ path: "2.realm", schema: this.realm.schema });
+      const object4 = realm2.write(() => {
+        return realm2.create<TestObject>("TestObject", { doubleCol: 1 });
+      });
+
+      expect(() => {
+        objects.indexOf(object4);
+      }).throws("Realm object is from another Realm");
+    });
+
+    it("implements addListener", function (this: RealmContext) {
+      //@ts-expect-error Navigator could be defined in some environments.
+      if (typeof navigator !== "undefined" && /Chrome/.test(navigator.userAgent)) {
+        // eslint-disable-line no-undef
+        // FIXME: async callbacks do not work correctly in Chrome debugging mode
+        return Promise.resolve();
+      }
```

Comment:
Since we don't support nor run these tests in chrome debugging mode anymore, this could probably just be deleted.
```suggestion
```